rust/connlib/tunnel/src/lib.rs

```diff
@@ -373,16 +384,22 @@ where
             _ => {}
         }
 
-        if let Poll::Ready(instant) = self.connection_pool_timeout.poll_unpin(cx) {
-            self.node.handle_timeout(instant);
-            if let Some(timeout) = self.node.poll_timeout() {
-                self.connection_pool_timeout = sleep_until(timeout).boxed();
-            }
+        Poll::Pending
+    }
 
-            cx.waker().wake_by_ref();
+    fn poll_timeout(&mut self, cx: &mut Context<'_>) -> Poll<()> {
+        if let Some(timeout) = self.node.poll_timeout() {
```

Comment:
The Node has its own timeout in addition to the ConnectionState?

What does the ConnectionState's timeout do?

Reply:
`Node`'s `poll_timeout` only indicates the next time `handle_timeout` needs to be called and will always return the same timeout until something changes in its internal state, it's not a future just a `Duration`.

`ConnectionState`'s timeout is a future, which is used to keep track of `Node`'s timeout and have the executor poll the `Tunnel` when it happens so we can send it down to `Node` to handle it.